src/Controllers/AngelTypesController.php

```diff
@@ -22,4 +53,108 @@ public function about(): Response
+            $data = $this->validate($request, [
+                'minutes' => 'required|int|min:1',
+            ]);
+            $minutes = (int) $data['minutes'];
+            $now = Carbon::now();
+
+            $key = $this->config->get('app_key');
+            $alg = $this->config->get('jwt_algorithm');
+
+            $payload = [
+                'sub' => 'join_angel_type',
+                'iat' => $now->timestamp,
+                'exp' => $now->addMinutes($minutes)->timestamp,
+                'id' => $angelType->id,
+                'by' => $this->auth->user()->id,
+            ];
+            $jwt = JWT::encode($payload, $key, $alg);
+            $qrData = url('/angeltypes/' . $angelType->id . '/join', ['token' => $jwt]);
+        }
+
+        return $this->response->withInput($data)->withView(
+            'pages/angeltypes/qr',
+            ['angel_type' => $angelType, 'qr_data' => $qrData],
+        );
+    }
+
+    public function join(Request $request): Response
+    {
+        $this->qrJoinEnabled();
+        $angelType = $this->getAngelType($request);
+
+        $jwt = $request->get('token', '');
+
+        $key = $this->config->get('app_key');
+        $alg = $this->config->get('jwt_algorithm');
+
+        try {
+            $decoded = JWT::decode($jwt, new Key($key, $alg));
+        } catch (BeforeValidException | ExpiredException) {
+            throw new HttpNotFound();
+        } catch (Exception $e) {
+            $this->log->error('JWT Error', ['exception' => $e]);
+            throw new HttpNotFound();
+        }
+
+        $type = $decoded->sub ?? null;
+        $id = $decoded->id ?? null;
+        if ($type !== 'join_angel_type' || $id !== $angelType->id) {
+            throw new HttpNotFound();
+        }
+
+        /** @var User $confirmingUser */
+        $confirmingUser = User::findOrFail($decoded->by ?? null);
+        /** @var UserAngelType $userAngelType */
+        $userAngelType = UserAngelType::firstOrNew([
+            'user_id' => $this->auth->user()->id,
+            'angel_type_id' => $angelType->id,
+        ]);
+
+        if (!$userAngelType->confirmUser) {
+            $userAngelType->confirmUser()->associate($confirmingUser);
+
+            $this->log->info(
+                'Joined angel type {type} ({type_id}) via QR code by {confirming_user} ({confirming_id})',
+                [
+                    'type' => $angelType->name,
+                    'type_id' => $angelType->id,
+                    'confirming_user' => $confirmingUser->name,
+                    'confirming_id' => $confirmingUser->id,
+                ]
+            );
+            $this->addNotification('angeltype.add.success');
+
+            $userAngelType->save();
+        }
+
+        return redirect(url('/angeltypes', ['action' => 'view', 'angeltype_id' => $angelType->id]));
+    }
+
+    protected function qrJoinEnabled(): void
+    {
+        if ($this->config->get('app_key') && $this->config->get('join_qr_code', true)) {
```

Comment:
I think it should be possible to deactivate this feature. I wouldn't associate app_key with this feature though.

Reply:
could you please add all used config options to the default config

Reply:
already implemented